sdk/react/panels/cast/CastDevicesScreen.js

```diff
@@ -0,0 +1,198 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import {
+  Animated,
+  ListView,
+  Text,
+  View,
+  TouchableHighlight,
+} from 'react-native';
+
+import {
+  BUTTON_NAMES,
+} from '../../constants';
+import Utils from '../../utils';
+
+import castScreenStyles from '../style/castScreenStyles.json';
+
+const styles = Utils.getStyles(castScreenStyles);
+
+let ds = new ListView.DataSource({ rowHasChanged: (r1, r2) => r1 === r2 });
+
+const dismissButtonSize = 20;
+const castButtonSize = 35;
+
+class CastDevicesScreen extends Component {
+  static propTypes = {
+    height: PropTypes.number.isRequired,
+    width: PropTypes.number.isRequired,
+    onDismiss: PropTypes.func.isRequired,
+    onDeviceSelected: PropTypes.func.isRequired,
+    config: PropTypes.object.isRequired,
+    deviceIds: PropTypes.array.isRequired,
+    deviceNames: PropTypes.array.isRequired,
+    selectedItem: PropTypes.string.isRequired,
+  };
+
+  componentWillMount() {
+    const { height, deviceNames } = this.props;
+
+    this.state = {
+      translateY: new Animated.Value(height),
+      opacity: new Animated.Value(0),
+      dataSource: ds.cloneWithRows(deviceNames),
```

Comment:
`dataSource` will not change if the component gets new `deviceNames`
take a look at these docs: https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
deleted as it doesn't used at all

Reply:
Was componentWillMount, became constructor, since it shouldn't be changed.

Reply:
there's no changes

Reply:
Wait for it)